### PR TITLE
Add documentation annotations for ctors, gtors, and stors in stormtypes (SYN-3444)

### PR DESCRIPTION
**Pull request by vEpiphyte**

### CODE REVIEW

File: synapse/lib/autodoc.py [hunk 1]
```diff
@@ -323,17 +327,35 @@ def docStormTypes(page, docinfo, linkprefix, islib=False, lvl=1,
...
             if isinstance(rtype, dict):
                 rname = rtype.get('type')
-                assert rname == 'function', f'Unknown type: loclname={loclname} rname={rname}'
+
+                if isinstance(rname, dict):
+                    raise AssertionError(f'rname as dict not supported loclname={loclname} rname={rname}')
+
+                isstor = False
+                isfunc = False
+                isgtor = False
```
Comment (rakuy0) [inline synapse/lib/autodoc.py:336]: Is `isgtor` used anywhere?

Comment (vEpiphyte) [reply synapse/lib/autodoc.py:336]: Currently it is not. Some of this is future proofing for more robust generation.

File: synapse/lib/stormtypes.py [hunk 2]
```diff
@@ -122,23 +122,120 @@ def _validateInfo(self, obj, info, name):
...
+        callsig = getCallSig(locl)
+        # Assert the callsig for a stor has one argument
+        callsig_args = [str(v).split('=')[0] for v in callsig.parameters.values()]
+        assert len(callsig_args) == 1, f'stor funcs must only have one argument for {obj} {info.get("name")}'
+
+    def _validateCtor(self, obj, info, name):
+        rtype = info.get('type')
+        funcname = rtype.pop('_ctorfunc')
+        locl = getattr(obj, funcname, None)
+        assert locl is not None, f'bad funcname=[{funcname}] for {obj} {info.get("name")}'
```
Comment (rakuy0) [inline synapse/lib/stormtypes.py:220]: Could we change this and the `_validateStor/Gtor` ones to start like `bad <ctor/gtor/stor>name` so it's more immediately apparent which failed?

File: synapse/lib/autodoc.py [hunk 3]
```diff
@@ -74,7 +74,7 @@
         'locals': {
             'type': 'array',
             'items': {'$ref': '#/definitions/stormtypeDoc'},
-            'description': 'A list of attributes or functions to document.',
+            'description': 'A list of attributes, functions, getters and setters to document.',
```
[the remark this answers is not in the document]
Comment (vEpiphyte) [reply synapse/lib/autodoc.py:77]: fixed

File: synapse/lib/stormtypes.py (hunk 2 above)
Comment (vEpiphyte) [reply synapse/lib/stormtypes.py:220]: yup!

**Review by rakuy0 — APPROVED** [review]

**Pull request merged**